_posts/2021/2021-01-25-memoize-expensive-code.md

```diff
@@ -0,0 +1,73 @@
+---
+title: "Memoize Expensive Code"
+description: "The ||= operator can provide significant performance improvements"
+layout: article
+category: ruby
+image:
+  base: '2021/memoize-expensive-code'
+  alt: "Red Floppy Disk"
+  credit:  Fredy Jacob
+  source: "https://unsplash.com/photos/t0SlmanfFcg"
+---
+
+Memoization is a performance optimization where the result of a slow or non-performant piece of code is temporarily stored and when the expensive code is called again, the stored value is returned.
+
+It is a specific form of caching that is relatively elegant to read and implement in Ruby due to the use of the `||=` (or equals) operator.
+
+
+## Instead of ...
+
+...repeating potentially expensive calculations:
+
+```ruby
+class OldTimeySweetShop
+  def average_sweets_per_jar
+    sweet_count / glass_jars.count
+  end
+
+  def sweet_count
+    glass_jars.sum do |jar|
+      jar.count_each_sweet_by_hand
+    end
+  end
+end
+```
+
+
+## Instead...
+
+```ruby
+class OldTimeySweetShop
+  # leave the same
```

Comment:
Maybe at least leave the `def average..` part? as `# leave the same` is a bit confusing on its own.